src/components/CompareResults/ResultsMain.tsx

```diff
@@ -91,18 +96,106 @@ function ResultsMain() {
     }
   }
 
+  /********** Edit Results Title Section **********/
+  const [rawSearchParams, updateRawSearchParams] = useRawSearchParams();
+  const initialComparisonTitle = rawSearchParams.get('title') ?? 'Results';
+  const [comparisonTitleName, setComparisonTitleName] = useState(
+    initialComparisonTitle,
+  );
+  const [previousComparisonTitle, setPreviousComparisonTitle] =
+    useState('Results');
+  const [editComparisonTitleInputVisible, showEditComparisonTitle] =
+    useState(false);
+
+  const handleEditInputToggle = () => {
+    if (editComparisonTitleInputVisible) {
+      setComparisonTitleName(previousComparisonTitle); //revert to previous title
+    } else {
+      setPreviousComparisonTitle(comparisonTitleName); //store current title as previous state
+    }
+
+    showEditComparisonTitle(!editComparisonTitleInputVisible);
+  };
+
+  const slugifyComparisonTitle = (title: string) => {
+    return (
+      title
+        /**
+         *please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize#examples
+         */
+        .normalize('NFD') // Normalize to decompose diacritics (e.g., é -> e)
+        .replace(/[\u0300-\u036f]/g, '') // Remove accents
+        .toLowerCase()
+        .replace(/[^a-z0-9]+/g, '-') // Convert non-alphanumeric to hyphen
+        .replace(/^-+|-+$/g, '') // Trim hyphens from start and end
+    );
+  };
+
+  const onComparisonTitleChange = (value: string) => {
+    //PR notes: fix for truncated letter at the end of entry
+    const slug = slugifyComparisonTitle(value);
+    rawSearchParams.set('title', slug);
```

Comment:
I would remove it, and just set it in `OnComparisonTitleSave` directly, someting like this:
```js
rawSearchParams.set("title", comparisonTitleName)
updateRawSearchParams(rawSearchParams)
```

Reply:
Delete this whole function?

Reply:
Ok, I see what you mean.

Reply:
sorry, yes, it was a comment just for this only line, I always forget that github doesn't show on which lines a comment applies despite I selected it or them before...